splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/sequence/AbstractSequence.java

```diff
@@ -52,6 +52,10 @@ public long getNext() throws StandardException{
         return currPosition.getAndAdd(incrementSteps);
     }
 
+    public long peekAtCurrentValue() throws StandardException {
+        return currPosition.get();
```

Comment:
Quoting from the docs:

> The SYSCS_UTIL.SYSCS_PEEK_AT_SEQUENCE function shows you the very next value that will be returned by a NEXT VALUE FOR clause. Users should never directly query the SYSSEQUENCES table, because that will cause sequence generator concurrency to slow drastically.

If the sequence has a state such that the next call to `getNext()` will allocate a new block, with the current change `peekAtCurrentValue()` will return `currPosition`, but that won't necessarily be the value that `getNext()` will return if called just afterwards, because the `currPosition` will be (possibly) updated by the new block allocation.